app-backend/db/src/main/scala/com/azavea/rf/database/OrganizationDao.scala

```diff
@@ -206,4 +215,35 @@ object OrganizationDao extends Dao[Organization] with LazyLogging {
       case None => throw new IllegalArgumentException(s"User not in database: ${subjectId}")
     }
   }
+
+  def addLogo(logoBase64: String, orgID: UUID, dataBucket: String): ConnectionIO[Organization] = {
+    val prefix = "org-logos"
+    val key = s"${orgID.toString()}.png"
+    val logoByte = ApacheBase64.decodeBase64(logoBase64)
+    val logoStream = new ByteArrayInputStream(logoByte)
+    val md = new ObjectMetadata()
+    val s3 = new AWSAmazonS3Client(new DefaultAWSCredentialsProviderChain)
+    val s3Client = new AmazonS3Client(s3)
+
+    md.setContentType("image/png")
+    md.setContentLength(logoByte.length)
+
+    if (s3Client.listKeys(dataBucket, prefix).contains(s"${prefix}/${key}")) {
+      s3Client.deleteObject(dataBucket, s"${prefix}/${key}")
```

Comment:
`PUT` overwrites contents of whatever key you've specified, so you don't have to check if it's there and delete it first -- it's just extra round trips again. But for buckets without too many things in them it shouldn't be very costly. Up to you if you want to remove.